test/test-indexable-repository.ts

```diff
@@ -159,6 +159,109 @@ describe("IndexableRepository", () => {
+                            });
+                            reaction.dispose();
+                            done();
+                        }
+                    });
+                });
+            });
+
+            describe("after the some entities have been cached", () => {
+                beforeEach(async () => {
+                    await repository.byIdAsync("entity1");
+                    await repository.byIdAsync("entity2");
+
+                    repository.mutableCopyById("batch1", "entity1");
+                    repository.mutableCopyById("batch1", "entity2");
+                    repository.mutableCopyById("batch2", "entity1");
+                });
+
+                describe("after updating the copies", () => {
+                    beforeEach(() => {
+                        const batch1entity1 = repository.mutableCopyById("batch1", "entity1");
+                        const batch1entity2 = repository.mutableCopyById("batch1", "entity2");
+                        const batch2entity1 = repository.mutableCopyById("batch2", "entity1");
+
+                        repository.setMutableCopy("batch1", { ...batch1entity1, value: "batch1entity1" });
+                        repository.setMutableCopy("batch1", { ...batch1entity2, value: "batch1entity2" });
+                        repository.setMutableCopy("batch2", { ...batch2entity1, value: "batch2entity1" });
+                    });
+
+                    test.each([["batch1", "entity1"], ["batch1", "entity2"], ["batch2", "entity1"]])(
+                        "%p in %p is properly updated",
+                        (batchId, entityId) =>
+                            expect(repository.mutableCopyById(batchId, entityId)).toStrictEqual({
+                                id: entityId,
+                                value: batchId + entityId
+                            })
+                    );
+
+                    describe("and discarding some changes", () => {
+                        beforeEach(() => {
+                            repository.discardMutableCopy("batch1", "entity1");
+                        });
+
+                        it("the mutable copy of entity1 in batch1 is reset", () =>
+                            expect(repository.mutableCopyById("batch1", "entity1")).toStrictEqual({
+                                id: "entity1",
+                                value: "value-entity1"
+                            })
+                        );
+
+                        test.each([["batch1", "entity2"], ["batch2", "entity1"]])(
+                            "%p in %p is still in the updated state",
+                            (batchId, entityId) =>
+                                expect(repository.mutableCopyById(batchId, entityId)).toStrictEqual({
+                                    id: entityId,
+                                    value: batchId + entityId
+                                })
+                        );
```

Comment:
Add a test: The original value is still present in the repository, no API calls are made.